js/src/ipc/message.ts

```diff
@@ -77,7 +80,7 @@ export class MessageReader implements IterableIterator<Message> {
         const buf = this.source.read(PADDING);
         const bb = buf && new ByteBuffer(buf);
         const len = +(bb && bb.readInt32(0))!;
-        return { done: len <= 0, value: len };
+        return { done: len === 0, value: len };
```

Comment:
Should we have some kind of assertion somewhere that verifies length > 0? What happens if a corrupt file has that value?

Reply:
That’s effectively handled in the subsequent call. We use the length read here as the number of bytes to consume from the stream, which should be the flatbuffer metadata.

If the length is incorrect, the flatbuffer metadata created from the buffer will be empty. If the length is too long, the stream will read as many bytes as it can, and close automatically. Both cases are interpreted by the stream reader as bad input, and ignored.